packages/governance/src/binaryBallotCounter.js

```diff
@@ -0,0 +1,130 @@
+// @ts-check
+
+import { assert, details as X } from '@agoric/assert';
+import { makeStore } from '@agoric/store';
+import { makePromiseKit } from '@agoric/promise-kit';
+import { Far } from '@agoric/marshal';
+
+import { E } from '@agoric/eventual-send';
+import { ChoiceMethod, buildBallot } from './ballotBuilder';
+
+const TIE_VOTE = "It's a tie!";
+
+const makeWeightedBallot = (ballot, weight) => ({ ballot, weight });
+
+const makeBinaryBallot = (question, positionAName, positionBName) => {
+  const positions = [];
+  assert.typeof(positionAName, 'string');
+  assert.typeof(positionBName, 'string');
+  positions.push(positionAName, positionBName);
+
+  return buildBallot(ChoiceMethod.CHOOSE_N, question, positions, 1);
+};
+
+// Exported for testing purposes
+const makeBinaryBallotCounter = (question, aName, bName) => {
+  const template = makeBinaryBallot(question, aName, bName);
+
+  assert(
+    template.getMethod() === ChoiceMethod.CHOOSE_N,
+    X`Binary ballot counter only works with CHOOSE_N`,
+  );
+  let isOpen = true;
+  const outcomePromise = makePromiseKit();
+  const tallyPromise = makePromiseKit();
+  const allBallots = makeStore('seat');
+
+  const getQuestionPositions = () => ({
+    question,
+    positionA: aName,
+    positionB: bName,
+  });
+
+  const recordBallot = (seat, filledBallot, weight = 1n) => {
+    allBallots.has(seat)
+      ? allBallots.set(seat, makeWeightedBallot(filledBallot, weight))
+      : allBallots.init(seat, makeWeightedBallot(filledBallot, weight));
+  };
+
+  const countVotes = async quorumChecker => {
+    assert(!isOpen, X`can't count votes while the election is open`);
+
+    // ballot template has position choices; Each ballot in allBallots should
+    // match. count the valid ballots and report results.
+    const [positionA, positionB] = template.getPositions();
```

Comment:
Are we getting these from the template because we don't trust aName and bName? Can we make that explicit?